tests/components/accessibilityControls-test.js

```diff
@@ -133,6 +134,147 @@ describe('AccessibilityControls', function () {
     accessibilityControls.handleKey(mockEvent)
   });
 
+  describe('API', function() {
+    var mockCtrl, a11yCtrls;
 
+    beforeEach(function() {
+      mockCtrl = {
+        state: {
+          accessibilityControlsEnabled: true,
+          screenToShow: CONSTANTS.SCREEN.PLAYING_SCREEN,
+          volumeState: {
+            volume: 0
+          }
+        },
+        skin: {
+          state: {
+            currentPlayhead: 0,
+            duration: 60
+          }
+        },
+        setVolume: function() {},
+        seek: function() {},
+        updateSeekingPlayhead: function() {}
+      };
+      a11yCtrls = new AccessibilityControls(mockCtrl);
+    });
+
+    describe('canSeek', function() {
+
+      it('should only allow seeking on playing, pause and end screens', function() {
+        for (var currentScreen in CONSTANTS.SCREEN) {
+          mockCtrl.state.screenToShow = currentScreen;
+          switch (currentScreen) {
+            case CONSTANTS.SCREEN.PLAYING_SCREEN:
+            case CONSTANTS.SCREEN.PAUSE_SCREEN:
+            case CONSTANTS.SCREEN.END_SCREEN:
+              expect(a11yCtrls.canSeek()).toBe(true);
+              break;
+            default:
+              expect(a11yCtrls.canSeek()).toBe(false);
+              break;
+          }
+        }
+      });
+
+      it ('should disable seeking when an ad is playing', function() {
+        mockCtrl.state.screenToShow = CONSTANTS.SCREEN.PLAYING_SCREEN;
+        mockCtrl.state.isPlayingAd = true;
+        expect(a11yCtrls.canSeek()).toBe(false);
+      });
+
+    });
+
+    describe('changeVolumeBy', function() {
+
+      it('should increase and decrease the volume', function() {
+        var volume = 0;
+        mockCtrl.setVolume = function(vol) {
+          volume = vol;
+        };
+        mockCtrl.state.volumeState.volume = 0;
+        a11yCtrls.changeVolumeBy(100, true);
+        expect(volume).toBe(1);
+        mockCtrl.state.volumeState.volume = 1;
+        a11yCtrls.changeVolumeBy(100, false);
+        expect(volume).toBe(0);
+        mockCtrl.state.volumeState.volume = 0.5;
+        a11yCtrls.changeVolumeBy(25, true);
+        expect(volume).toBe(0.75);
+      });
+
+      it('should not call controller.setVolume() when requested change results in current volume', function() {
+        var setVolumeCalled = false;
+        mockCtrl.setVolume = function() {
+          setVolumeCalled = true;
+        };
+        mockCtrl.state.volumeState.volume = 0;
+        a11yCtrls.changeVolumeBy(100, false);
+        expect(setVolumeCalled).toBe(false);
+      });
+
+      it('should constrain volume to supported values', function() {
+        var volume = 0;
+        mockCtrl.setVolume = function(vol) {
+          volume = vol;
+        };
+        mockCtrl.state.volumeState.volume = 0.5;
+        a11yCtrls.changeVolumeBy(500, true);
```

Comment:
Can we add a check to see if negative volumes are ignored as well?

Reply:
Will do